### PR TITLE
Add the spec v2 draft

### PR DESCRIPTION
**Pull request by chee**
have a wee look
have a little think about biggly winks

### CODE REVIEW

File: _specification-v2/components.md [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+</aside>
+
+## HTML
+
+- A component’s markup must be contained in a root element that must:
+	- Have a data-o-component attribute with value of the component’s name, or
+	- Be defined as a
+	  [customElement](https://html.spec.whatwg.org/multipage/custom-elements.html#custom-element)
+	  registered against the component’s class
+- All data attributes and IDs must be namespaced with the component’s name.
```
Comment (JakeChampion) [inline _specification-v2/components.md:105]: Adding the boldness on the `must`s, do we need to add boldness the the `or` at the end of the first bullet point?
```suggestion
- A component’s markup **must** be contained in a root element that **must**:
	- Have a data-o-component attribute with value of the component’s name, or
	- Be defined as a
	  [customElement](https://html.spec.whatwg.org/multipage/custom-elements.html#custom-element)
	  registered against the component’s class
- All data attributes and IDs **must** be namespaced with the component’s name.
```

File: _specification-v2/components.md [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+- CSS selectors should follow the BEM naming convention.
+- Sass functions, variables, mixins and placeholders defined the global
+  namespace **must** be prefixed with the component name
+- A component **must** only style:
+	- The component’s root element and its children
+	- Elements with classes prefixed with the component’s name, and their
+	  children
+	- Elements with any other data-attribute in the component’s namespace, and
+	  their children
+- A component must not output CSS by default when Sass is loaded (i.e. `@use`, or
```
Comment (JakeChampion) [inline _specification-v2/components.md:130]:
```suggestion
- A component **must not** output CSS by default when Sass is loaded (i.e. `@use`, or
```

File: _specification-v2/manifest.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+		<td><code>String</code> or <code>null</code></td>
+	</tr>
+	<tr>
+		<th scope="row" role="rowheader">Required</th>
+		<td><code>true</code></td>
+	</tr>
+</table>
+
+Defines the type of Origami project that the manifest belongs to. **Must** be
+set to a string:
```
Comment (JakeChampion) [inline _specification-v2/manifest.md:65]: we need to mention it an be set to `null`, which is also in the list of possible options
For the code suggestion, I copied the text from the `status` part of this spec which uses "must be one of:"
```suggestion
Defines the type of Origami project that the manifest belongs to. **Must** be one of:
```

**Review by JakeChampion — APPROVED** [review]
This is a nice write up of the spec, it was a joy to read

File: _specification-v2/manifest.md (hunk 3 above)
Comment (chee) [reply _specification-v2/manifest.md:65]: oh yes, this was something I wanted to talk about.
I'm wondering if we should say something like:

> **must** be set to a string, .e.g.:

and then remove `null` as an option. `type` then becomes an entry point for extensions. the component spec augments the spec by saying "when the `type` is "component" then all of this applies:" and you opt into this set of rules by setting the type to that

Comment (JakeChampion) [reply _specification-v2/manifest.md:65]: I like that idea a lot.

I checked what sets it to null, and the list is:
- change-api-action -- github action
- origami-apply-labels -- github action
- origami-labels -- github action
- origami-percy -- github action
- origami-project-board-action -- github action
- origami-version -- github action
- polyfill-library-node -- npm libary
- useragent_parser -- npm library

Comment (chee) [reply _specification-v2/manifest.md:65]:
```suggestion
Defines the type of Origami project that the manifest belongs to. **Must** be set to a string, e.g.:
```

File: _specification-v2/manifest.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+- `"component"`: A front-end component that follows [the component specification](/spec/v2/components/)
+- `"imageset"`: A set of images that have an alias on the Origami Image Service
+- `"service"`: An HTTP service
+- `"cli"`:	A command line tool
+- `"library"`:	A library that is not a front-end component
+- `"website"`: Origami websites that aren't intended to be services
+- `"config"`: Projects that are configuration for other projects
+- `"example"`: Example and boilerplate projects
+- `"meta"`: Repository-only projects that relate to how Origami works
```
Comment (chee) [inline _specification-v2/manifest.md:74]:
```suggestion
- `"action"`: [A Github Action](https://github.com/features/actions)
- `"meta"`: Repository-only projects that relate to how Origami works
```

File: _specification-v2/manifest.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+- `null`: An Origami project that does not fit any of the named categories
```
Comment (chee) [inline _specification-v2/manifest.md:75]:
```suggestion
```

File: _specification-v2/manifest.md (hunk 3 above)
Comment (chee) [reply _specification-v2/manifest.md]: nice work! I've added a description of `"action"` to this list, we can update those to use that. and maybe even we write a spec for our github actions some time:)

**Review by JakeChampion — APPROVED** [review]
🎉

**Review by notlee — COMMENTED** [review]
Reviewing the spec in one sitting. Excellent and exciting :)

File: _specification-v2/components.md [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+- Elements with any other data attribute in the component’s namespace, and their
+  children
+- The component **must** be configurable using data attributes
+- Any events triggered by a component **must** be namespaced under the target
+  component’s name
+- Component JavaScript **should** be annotated (e.g. using JSDoc)
+
+## Sass
+
+- CSS selectors **should** follow the BEM naming convention.
```
[the remark this answers is not in the document]
Comment (notlee) [reply _specification-v2/components.md]: that's better 👍

**Review by notlee — APPROVED** [review]

File: _specification-v2/components.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (chee) [reply _specification-v2/components.md]: thanks for the suggestion! :)

**Comment by chee** [comment]
note to self: add an ` .o-message--warning-light` to the v1 of the component spec :)

**Comment by JakeChampion** [comment]
I tried to find these new pages on the branch deployment and I couldn't.
I think the pages need to be added to a jekyll collection to end up being created as part of the built website
https://github.com/Financial-Times/origami-website/blob/396a8c81aa5ebee2ee3096db394b024fd1f19f4a/_config.yml#L3-L21

Should that happen in a separate pull-request or as part of this one?

**Comment by notlee** [comment]
Can we close this since we decided not to continue with v2? I've opened a new PR for us to update the site for the npm milestone here, all in once place: https://github.com/Financial-Times/origami-website/pull/360

**Pull request closed without merging**